### PR TITLE
Stats: Add Subscribers `Emails Opens and Clicks` Details

### LINKED ISSUE #23045: Subscribers Email Opens and Clicks Card & Details
Full Scope: https://github.com/Automattic/wordpress-mobile/issues/47
Design: [IqhXWz3Iir7RMb5XH5gGfZ-fi-883%3A4006](https://href.li/?https://www.figma.com/file/IqhXWz3Iir7RMb5XH5gGfZ/?node-id=883%3A4006)

Create Emails Opens and Clicks card.

#### More Details

This task will likely require the implementation of new endpoints that are used on the web.

(`/rest/v1.1/sites/{site_id}/stats/emails/summary?quantity=30&sort_field=opens&sort_order=desc`) fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Srznvy%2Qfgngf%2Srznvy%2Qfhzznel.cuc%3Se%3Q53087o60%26sv%3Qtrg_rznvy_fhzznel%2335-og

### Scope 

- Emails Opens and Clicks Card that displays 6 item list
- Emails Opens and Clicks Details that display a longer list as on WP.com

### PR DESCRIPTION
**Pull request by staskus**
Fixes #23045
Continuation of https://github.com/wordpress-mobile/WordPress-iOS/pull/23056 PR

Add navigation from Subscribers `Emails Opens and Clicks` card into `Details` that displays more information.

## Solution

I decided to integrate within `SiteStatsDetailTableViewController` which is used by both Insights, and Traffic details for most of the Stat sections.

`SiteStatsDetailTableViewController` requires to comply by providing loading, error, and loaded states for a supported `StatSection`. It was fairly straightforward to do by reusing `StatsSubscribersStore`.

## To test:

1. Enable `Stats Traffic and Subscibers tab` feature flag
2. Select a high-traffic site
3. Open Stats
4. Open Subscribers
5. Confirm `Emails Opens and Clicks` card appears with "View more" button
6. Select "View more"
7. Confirm a longer `Emails Opens and Clicks` list appears in details
8. Compare the data with WP.com and confirm that it's the same data

## Regression Notes
1. Potential unintended areas of impact

None, not affecting any existing functionality

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailsViewModel.swift [hunk 1]
```diff
@@ -392,6 +418,10 @@ class SiteStatsDetailsViewModel: Observable {
         }
         ActionDispatcher.dispatch(PeriodAction.refreshPeriod(query: .postStats(postID: postID)))
     }
+
+    func refreshEmailsSummary() {
+        subscribersStore.updateEmailsSummary(quantity: 30, sortField: .opens)
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailsViewModel.swift:423]: I've chosen the same quantity and sorting as on the web for now.

Comment (guarani) [reply WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailsViewModel.swift:423]: nitpick: I think replicating the web here is safe, but I wonder why it was done this way. For example:
1. I see six emails on the Subscriber tab
2. I click "View more" to see the rest
3. The Emails screen shows emails, but it's not clear which are the ones I haven't seen on the previous screen

I'm not sure why the web would intentionally sort by post ID on one page and by opens on the next. The latter seems much more useful on both screens.

**Review by guarani — APPROVED** [review]
Works great! Thanks @staskus 🙌

**Comment by staskus** [comment]
> I'm not sure why the web would intentionally sort by post ID on one page and by opens on the next. The latter seems much more useful on both screens.

That's a fair point, @guarani. I'll try to get clarification from the web team. It's an easy change we can make.

**Pull request merged**